File: app/src/main/java/in/smslite/viewModel/BlockedMessageActivityViewModel.java
package in.smslite.viewModel;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;

import java.util.List;

import in.smslite.db.Message;
import in.smslite.db.MessageDatabase;

/**
 * Created by rahul1993 on 5/31/2018.
 */

public class BlockedMessageActivityViewModel extends AndroidViewModel {
  private MessageDatabase mDB;
  public BlockedMessageActivityViewModel(@NonNull Application application) {
    super(application);
    mDB = MessageDatabase.getInMemoryDatabase(application);
  }

  public void updateCategoryOfMessage(String address, int category, int presentCategory){
    mDB.messageDao().moveToCategory(address, category, presentCategory);
  }

  public LiveData<List<Message>> getBlockedMessage(){
    return mDB.messageDao().getBlockedMessage();
  }
}

File: app/src/main/java/in/smslite/viewModel/BlockedMessageActivityViewModel.kt
package `in`.smslite.viewModel

import `in`.smslite.db.Message
import android.app.Application
import androidx.lifecycle.LiveData
import kotlinx.coroutines.launch

/**
 * Created by rahul1993 on 5/31/2018.
 */
class BlockedMessageActivityViewModel(application: Application) : BaseViewModel(application) {
    fun updateCategoryOfMessage(address: String?, category: Int, presentCategory: Int) {
        ioScope.launch {
            messageRepository.moveToCategory(address, category, presentCategory)
        }
    }

    val blockedMessage: LiveData<List<Message>>
        get() = messageRepository.getBlockedMessage()
}